### PR TITLE
Update enum definitions / dependencies

### PR DESCRIPTION
**Pull request by sublimator**

### CODE REVIEW

**Comment by sublimator** [comment]
@shekenahglory

File: package.json [hunk 1]
```diff
@@ -12,26 +12,28 @@
...
-    "eslint": "^1.10.3",
-    "fs-extra": "^0.24.0",
+    "eslint": "^2.9.0",
+    "fs-extra": "^0.30.0",
     "intercept-stdout": "^0.1.2",
-    "istanbul": "~0.3.5",
-    "mocha": "~2.1.0",
+    "istanbul": "~0.4.3",
+    "mocha": "~2.4.5",
     "ripple-lib": "^0.12.9"
```
Comment (clark800) [inline package.json]: Looks like we can remove the `ripple-lib` dependency now

File: src/quality.js [hunk 2]
```diff
@@ -16,7 +15,7 @@ module.exports = {
   decode(arg) {
     const bytes = slice(parseBytes(arg), -8);
     const exponent = bytes[0] - 100;
-    const mantissa = new Decimal(bytesToHex(slice(bytes, 1)), 16);
+    const mantissa = new Decimal('0x' + bytesToHex(slice(bytes, 1)), 16);
```
Comment (clark800) [inline src/quality.js]: Looks like the `Decimal` constructor only takes one argument as of v5.0

**Comment by clark800** [comment]
`rm -rf node_modules; npm install` fails for me on OSX during the build of `bufferutil` and `utf-8-validate`. Subsequently `npm test` errors with `Error: Cannot find module 'ripple-lib'`.

File: package.json (hunk 1 above)
Comment (sublimator) [inline package.json]: Yes, may as well, I don't think it's actually being used anyway

File: src/quality.js (hunk 2 above)
Comment (sublimator) [inline src/quality.js]: Yeah, oops

**Comment by sublimator** [comment]
@clark800 updated

**Comment by clark800** [comment]
When I try `npm link`ing this into ripple-lib, it causes the ripple-lib unit tests to fail. I'm not sure if it's related, but the `npm link` on the ripple-binary-codec fails due to lint errors, but it looks like the symlink is in place.

**Comment by sublimator** [comment]
Conflicting eslint versions ?

**Comment by sublimator** [comment]
Which tests fail?

It's late here. Feel free to take over this (@ anyone), or I'll look into it tomorrow.

**Comment by clark800** [comment]
> Which tests fail?

It doesn't even run the tests, just
```
Transformation error; return original code
{ [Error: Line 1: Unexpected token ILLEGAL]
  lineNumber: 1,
  description: 'Unexpected token ILLEGAL',
  index: 0 }
npm ERR! Test failed.  See above for more details.
```
All but the last line are normal, the illegal token doesn't usually cause a failure.

**Comment by sublimator** [comment]
I also get the same message when doing an `npm link` as you described. I don't however when running `./node_modules/mocha/bin/_mocha` without the `isparta`

**Comment by clark800** [comment]
That's right, `isparta` eats error messages; here is the error I get without `isparta`:

```/Local/Users/clark/ripple/ripple-binary-codec/distrib/npm/enums/index.js:120
      return _.indexBy(fields, 'name');},
               ^

TypeError: _.indexBy is not a function
at Function.valuesByName (index.js:120:16)```

**Comment by sublimator** [comment]
That should be keyBy now in lodash 4 I guess. I thought I got to that ??

**Comment by sublimator** [comment]
See: https://github.com/douglasduteil/isparta/issues/68#issuecomment-191752598

`--root src/`

**Comment by sublimator** [comment]
Are you by chance on the wrong commit somehow ?

**Comment by sublimator** [comment]
Maybe we want to back off all the dependencies except the eslint ? ripple-lib is using lodash 3.x it seems

**Comment by clark800** [comment]
I'm on `dd1b64f`. Ah, it looks like the problem was that `npm run compile` never executed because the lint check failed. I ran it manually, so now there is one failure in the unit tests: the "combine" test:

```
  1) RippleAPI combine:
     TypeError: Cannot read property 'from' of undefined
      at st-object.js:32:25
      at /Local/Users/clark/ripple/ripple-binary-codec/node_modules/lodash/lodash.js:13057:16
      at /Local/Users/clark/ripple/ripple-binary-codec/node_modules/lodash/lodash.js:4389:15
      at baseForOwn (/Local/Users/clark/ripple/ripple-binary-codec/node_modules/lodash/lodash.js:2652:24)
      at Function.transform (/Local/Users/clark/ripple/ripple-binary-codec/node_modules/lodash/lodash.js:13056:39)
      at Function.from (st-object.js:29:18)
      at serializeObject (binary.js:22:18)
      at Object.encode (index.js:17:21)
      at RippleAPI.combine (combine.js:34:36)
      at Context.<anonymous> (api-test.js:369:31)
      at run (node_modules/babel-polyfill/node_modules/core-js/modules/es6.promise.js:104:47)
      at node_modules/babel-polyfill/node_modules/core-js/modules/es6.promise.js:115:28
      at flush (node_modules/babel-polyfill/node_modules/core-js/modules/$.microtask.js:19:5)
```

**Comment by clark800** [comment]
Ok, all I had to do was delete the cached `eslintrc` file!

**Comment by sublimator** [comment]
Looks like transform in lodash 4 is getting things like `toString` ??

**Comment by sublimator** [comment]
This "fixes" that test:
```
      if (typeof value === 'object') {
        return _.transform(value, (so, val, key) => {
          const field = Field.hasOwnProperty(key) && Field[key];
```

I'm wary of what other changes are lurking, so we should probably back lodash off to 3.x

**Comment by sublimator** [comment]
Do the `Connection` tests always take so long ?

**Comment by sublimator** [comment]
ha! 
```
npm WARN unmet dependency /Users/ndudfield/ripple-binary-codec/node_modules/eslint requires lodash@'^4.0.0' but will load
npm WARN unmet dependency /Users/ndudfield/ripple-binary-codec/node_modules/lodash,
npm WARN unmet dependency which is version 3.10.1
```

**Comment by sublimator** [comment]
Do we feel like updating lodash in ripple-lib ? :)

**Comment by clark800** [comment]
> Do we feel like updating lodash in ripple-lib ? :)

Is there any way to make it work without updating?

**Comment by sublimator** [comment]
I'm not sure. I haven't been doing much js stuff lately

**Comment by clark800** [comment]
> I'm wary of what other changes are lurking, so we should probably back lodash off to 3.x

I think if you just add a unit test to cover this case that caused the "combine" test to fail it should be fine to stick with 4.x and it should be good to merge

**Comment by sublimator** [comment]
Up 2 u

**Comment by sublimator** [comment]
@clark800 updated

**Comment by clark800** [comment]
LGTM after squash

**Pull request merged**